mps-cli-gradle-plugin/plugin/src/main/groovy/org/mps_cli/cone_of_influence/Filesystem2SSolutionBridge.groovy

```diff
@@ -8,29 +13,83 @@ import static groovy.io.FileType.FILES
 
 class Filesystem2SSolutionBridge {
 
-    /** Returns a list of .msd Files representing the modified modules. **/
-    static HashSet<Path> computeModulesWhichAreModifiedInCurrentBranch(String gitRepoRootLocation, List<String> allModifiedFiles) {
-        def gitRepoLocation = Paths.get(gitRepoRootLocation).toAbsolutePath().normalize()
-
-        LinkedHashSet<Path> affectedModulesFiles = []
-        allModifiedFiles.each {
-            if (it.endsWith('.msd'))
-                affectedModulesFiles.add(gitRepoLocation.resolve(it))
-            else {
-                def filterSolutionFiles = ~/.*\.msd$/
-                def myFile = gitRepoLocation.resolve(it).parent
-                def affectedModuleFound = false
-                while (myFile != null) {
-                    if (Files.exists(myFile)) {
-                        myFile.traverse(type: FILES, maxDepth: 0, nameFilter: filterSolutionFiles) {
-                            affectedModuleFound = true; affectedModulesFiles.add(it) }
+    SRepository repository
+
+    private static def modulePattern = ~/.*\.msd/
+    private static def modelPattern = ~/.*\.mps|.*\.model/
```

Comment:
the model patter should contain mpsr filex (root nodes saved in file per root persistency)

... at best we look if files changed inside the "models" directory ... (due to custom persistency it can be the case that we have root nodes also with other extensions) 


The layout of a directory with file-per-root persistency is "solution_name/models/model_name/(.model|rootname.mpsr|custom_persistency)"

Basically if the changes are inside a directory "models" then the model changed,

Reply:
I think this case is already covered. If an `*.mpsr`/custom persistency file was modified, then we detect with `modelPattern` that there is a `.model` file in the same folder, so we know that a model was modified.